lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -68,6 +68,20 @@ bool isTransformChange(const TfToken& nameToken)
     return nameToken == UsdGeomTokens->xformOpOrder || UsdGeomXformOp::IsXformOp(nameToken);
 }
 
+// Prevent exception from the notifications from escaping and breaking USD/Maya.
+// USD does not wrap its notification in try/catch, so we need to do it ourselves.
+template <class RECEIVER> struct SafeNotification
```

Comment:
Can be just a single templated function, doesn't need to be static, we're in the unnamed namespace already:
template<class R, class N> void notify(const N& n) {
    // try / catch stuff
    R::notify(n);
}
You can then explicitly specify a single template argument and the second template argument (the type of the function argument) will be deduced by the compiler.

Reply:
Ah good, I thought that if we provided one template type, we had to provide both, that is why I use this trick to split them.